### PR TITLE
Fix: mmap len cannot be zero

### LINKED ISSUE #1172: Reachable assertion in `get_intersected_range()` by invalid `size` of `mmap`/`mprotect`/`munmap`/`madvise`
### Describe the bug
There is a reachable assertion in `get_intersected_range()` at kernel/aster-nix/src/vm/vmar/mod.rs:844 when make a `mmap` syscall with specific arguments.

https://github.com/asterinas/asterinas/blob/4844e7ca7ca6d78896a51a71487a6fdfe9ca6654/kernel/aster-nix/src/vm/vmar/mod.rs#L841-L846

### To Reproduce
1. Compile a program which calls `mmap` with specific arguments:
```C
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/mman.h>
#include <sys/wait.h>

int main(void)
{
  int fd = open("/proc/meminfo", O_RDONLY);
  mmap(NULL, 0, 3, 460833, fd, 0);
  return 0;
}

```
2. Run the compiled program in Asterinas.

### Expected behavior
Asterinas reports assertion failure and is terminated.

### Environment

- Official docker asterinas/asterinas:0.6.2
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 4844e7c

### Logs

```
~ # /root/mmap_test 
panicked at /root/asterinas/kernel/aster-nix/src/vm/vmar/mod.rs:844:5:
assertion failed: is_intersected(range1, range2)
Printing stack trace:
   1: fn 0xffffffff88837ff0 - pc 0xffffffff88838008 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4a8b10;

   2: fn 0xffffffff88837dd0 - pc 0xffffffff88837f48 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4a8b20;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4a8ca0;

   4: fn 0xffffffff8895ce70 - pc 0xffffffff8895cef2 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4a8cb0;

   5: fn 0xffffffff8895d010 - pc 0xffffffff8895d050 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4a8d40;

   6: fn 0xffffffff883fdc90 - pc 0xffffffff883fdcc9 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4a8db0;

   7: fn 0xffffffff883fafa0 - pc 0xffffffff883fbd1c / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4a8de0;

   8: fn 0xffffffff8804ba60 - pc 0xffffffff8804bc05 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4a98d0;

   9: fn 0xffffffff8804c350 - pc 0xffffffff8804c4f4 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4a9cd0;

  10: fn 0xffffffff88516010 - pc 0xffffffff88516c98 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4aa000;

  11: fn 0xffffffff88515d50 - pc 0xffffffff88515e95 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4aa9b0;

  12: fn 0xffffffff88407d50 - pc 0xffffffff8840c8ae / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4aab40;

  13: fn 0xffffffff8834bd20 - pc 0xffffffff8834bdae / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4c03d0;

  14: fn 0xffffffff88520700 - pc 0xffffffff8852127f / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4c0570;

  15: fn 0xffffffff88075a80 - pc 0xffffffff88075a8e / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4c0f90;

  16: fn 0xffffffff8882cba0 - pc 0xffffffff8882cbb6 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4c0fb0;

  17: fn 0xffffffff88798ad0 - pc 0xffffffff88798b39 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4c0fd0;

  18: fn                0x0 - pc                0x0 / registers:

     rax               0x12; rdx 0xffffffff889edf38; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f4c1000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:107
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) au2ndt6r5ksv2yyhocg58xyv7:?
(  4) ??:?
(  5) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:220
(  6) ??:?
(  7) /root/asterinas/kernel/aster-nix/src/vm/vmar/mod.rs:661
(  8) /root/asterinas/kernel/aster-nix/src/vm/vmar/vm_mapping.rs:88
(  9) /root/asterinas/kernel/aster-nix/src/vm/vmar/vm_mapping.rs:741
( 10) /root/asterinas/kernel/aster-nix/src/syscall/mmap.rs:89
( 11) /root/asterinas/kernel/aster-nix/src/syscall/mmap.rs:29
( 12) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:215
( 13) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:323
( 14) /root/asterinas/kernel/aster-nix/src/thread/task.rs:69
( 15) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 16) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 17) /root/asterinas/ostd/src/task/task.rs:295
make: *** [Makefile:153: run] Error 1
```

### LINKED ISSUE #1184: Reachable assert in `sys_mprotect()` by invalid `addr`
### Describe the bug

There is a reachable assert caused by invalid `addr` value in `sys_mprotect()` at kernel/aster-nix/src/syscall/mprotect.rs:15 when make a `mprotect` syscall with specific arguments.
https://github.com/asterinas/asterinas/blob/0291b5dc6bb142b9c6165c1cb29b7658eefdaa63/kernel/aster-nix/src/syscall/mprotect.rs#L15

### To Reproduce
1. Compile a program which calls `mprotect` with specific arguments:
```C
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/mman.h>
#include <errno.h>

int main() {
  mprotect(0xffffffff, 0, 0x5);

  return EXIT_SUCCESS;
}
```
2. Run the compiled program in Asterinas.

### Expected behavior
Asterinas reports panic and is terminated.

### Environment
- Official docker asterinas/asterinas:0.6.2
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 0291b5dc

### Logs
```
~ # /root/mprotect.c 
panicked at /root/asterinas/kernel/aster-nix/src/syscall/mprotect.rs:15:5:
assertion failed: addr % PAGE_SIZE == 0
Printing stack trace:
   1: fn 0xffffffff887ccd10 - pc 0xffffffff887ccd28 / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2aa660;

   2: fn 0xffffffff887ccaf0 - pc 0xffffffff887ccc68 / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2aa670;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2aa7f0;

   4: fn 0xffffffff8895bbd0 - pc 0xffffffff8895bc52 / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2aa800;

   5: fn 0xffffffff8895bd70 - pc 0xffffffff8895bdb0 / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2aa890;

   6: fn 0xffffffff884e8350 - pc 0xffffffff884e862d / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2aa900;

   7: fn 0xffffffff88519340 - pc 0xffffffff8851e37b / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2aab40;

   8: fn 0xffffffff8815cab0 - pc 0xffffffff8815cb3e / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2c03d0;

   9: fn 0xffffffff88567100 - pc 0xffffffff88567c7f / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2c0570;

  10: fn 0xffffffff880c9ef0 - pc 0xffffffff880c9efe / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2c0f90;

  11: fn 0xffffffff887a7130 - pc 0xffffffff887a7146 / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2c0fb0;

  12: fn 0xffffffff88843a00 - pc 0xffffffff88843a69 / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2c0fd0;

  13: fn                0x0 - pc                0x0 / registers:

     rax               0x12; rdx 0xffffffff889e5488; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2c1000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:107
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) 86161ztg45fpd9b1tjbn4gsl1:?
(  4) ??:?
(  5) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:220
(  6) ao06edr9di1q0vhz8iz1v26bt:?
(  7) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:166
(  8) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:323
(  9) /root/asterinas/kernel/aster-nix/src/thread/task.rs:69
( 10) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 11) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 12) /root/asterinas/ostd/src/task/task/mod.rs:310
make: *** [Makefile:153: run] Error 1
```

### PR DESCRIPTION
**Pull request by StevenJiang1110**
Fix #1172 
Fix #1184

### CODE REVIEW

**Comment by Marsman1996** [comment]
Hi, I can trigger assertion in get_intersected_range() through `mprotect` too.

Here is the syscall program.
```C
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/mman.h>
#include <errno.h>

int main() {
    void *addr = mmap(NULL, 4096, 0x3, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
    mprotect(addr, 0, 0x5);

    return EXIT_SUCCESS;
}

```

And here is the log:
```
~ # /root/mprotect.c 
panicked at /root/asterinas/kernel/aster-nix/src/vm/vmar/mod.rs:773:5:
assertion failed: is_intersected(range1, range2)
Printing stack trace:
   1: fn 0xffffffff887ccd40 - pc 0xffffffff887ccd58 / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a100;

   2: fn 0xffffffff887ccb20 - pc 0xffffffff887ccc98 / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a110;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a290;

   4: fn 0xffffffff8895bc00 - pc 0xffffffff8895bc82 / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a2a0;

   5: fn 0xffffffff8895bda0 - pc 0xffffffff8895bde0 / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a330;

   6: fn 0xffffffff884f6400 - pc 0xffffffff884f6439 / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a3a0;

   7: fn 0xffffffff884f0920 - pc 0xffffffff884f0f6d / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a3d0;

   8: fn 0xffffffff884f0710 - pc 0xffffffff884f07f2 / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a730;

   9: fn 0xffffffff88161df0 - pc 0xffffffff88161e9f / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a840;

  10: fn 0xffffffff884e8350 - pc 0xffffffff884e865d / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a900;

  11: fn 0xffffffff88519370 - pc 0xffffffff8851e3ab / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22ab40;

  12: fn 0xffffffff8815cab0 - pc 0xffffffff8815cb3e / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2403d0;

  13: fn 0xffffffff88567130 - pc 0xffffffff88567caf / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f240570;

  14: fn 0xffffffff880c9ef0 - pc 0xffffffff880c9efe / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f240f90;

  15: fn 0xffffffff887a7160 - pc 0xffffffff887a7176 / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f240fb0;

  16: fn 0xffffffff88843a30 - pc 0xffffffff88843a99 / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f240fd0;

  17: fn                0x0 - pc                0x0 / registers:

     rax               0x12; rdx 0xffffffff889e5500; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f241000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:107
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) 86161ztg45fpd9b1tjbn4gsl1:?
(  4) ??:?
(  5) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:220
(  6) ??:?
(  7) /root/asterinas/kernel/aster-nix/src/vm/vmar/mod.rs:256
(  8) /root/asterinas/kernel/aster-nix/src/vm/vmar/mod.rs:237
(  9) /root/asterinas/kernel/aster-nix/src/vm/vmar/static_cap.rs:111
( 10) /root/asterinas/kernel/aster-nix/src/syscall/mprotect.rs:18
( 11) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:166
( 12) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:323
( 13) /root/asterinas/kernel/aster-nix/src/thread/task.rs:69
( 14) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 15) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 16) /root/asterinas/ostd/src/task/task/mod.rs:310
make: *** [Makefile:153: run] Error 1
```

**Comment by StevenJiang1110** [comment]
> Hi, I can trigger assertion in get_intersected_range() through mprotect too.

I check [the behavior of mprotect in linux](https://elixir.bootlin.com/linux/v6.0.9/source/mm/mprotect.c#L683), the problem is also due to we don't handle length zero here. I will fix it in this PR.

**Comment by Marsman1996** [comment]
It seems that we also need to check the `size` of `munmap()`:

<details>

<summary> Syscall code </summary>

```C
#include <errno.h>
#include <stdio.h>
#include <stdlib.h>
#include <sys/mman.h>
#include <unistd.h>

int main() {
  void *addr = mmap(NULL, 4096, 0x3, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
  mprotect(addr, 0, 0x5);
  munmap(addr, 0);

  return EXIT_SUCCESS;
}

```
</details>





<details>

<summary> log info </summary>

```
~ # /root/mprotect.c 
0
panicked at /root/asterinas/kernel/aster-nix/src/vm/vmar/mod.rs:364:13:
assertion failed: is_intersected(&vm_mapping_range, &range)
Printing stack trace:
   1: fn 0xffffffff887ccda0 - pc 0xffffffff887ccdb8 / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a0e0;

   2: fn 0xffffffff887ccb80 - pc 0xffffffff887cccf8 / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a0f0;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a270;

   4: fn 0xffffffff8895bc60 - pc 0xffffffff8895bce2 / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a280;

   5: fn 0xffffffff8895be00 - pc 0xffffffff8895be40 / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a310;

   6: fn 0xffffffff88087ce0 - pc 0xffffffff880885fc / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a380;

   7: fn 0xffffffff88083330 - pc 0xffffffff88083380 / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a810;

   8: fn 0xffffffff88518cb0 - pc 0xffffffff8851910f / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22a860;

   9: fn 0xffffffff88120c30 - pc 0xffffffff88126144 / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f22ab40;

  10: fn 0xffffffff880e80c0 - pc 0xffffffff880e814e / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f2403d0;

  11: fn 0xffffffff880a1f70 - pc 0xffffffff880a2aef / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f240570;

  12: fn 0xffffffff88441850 - pc 0xffffffff8844185e / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f240f90;

  13: fn 0xffffffff887a71c0 - pc 0xffffffff887a71d6 / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f240fb0;

  14: fn 0xffffffff88843a90 - pc 0xffffffff88843af9 / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f240fd0;

  15: fn                0x0 - pc                0x0 / registers:

     rax               0x12; rdx 0xffffffff889e54b0; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007f241000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:107
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) 86161ztg45fpd9b1tjbn4gsl1:?
(  4) ??:?
(  5) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:220
(  6) /root/asterinas/kernel/aster-nix/src/vm/vmar/mod.rs:364
(  7) /root/asterinas/kernel/aster-nix/src/vm/vmar/static_cap.rs:131
(  8) /root/asterinas/kernel/aster-nix/src/syscall/munmap.rs:13
(  9) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:159
( 10) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:323
( 11) /root/asterinas/kernel/aster-nix/src/thread/task.rs:69
( 12) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 13) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 14) /root/asterinas/ostd/src/task/task/mod.rs:310
```
</details>

**Comment by StevenJiang1110** [comment]
> It seems that we also need to check the size of munmap()

The problem is fixed.

File: kernel/aster-nix/src/syscall/mprotect.rs [hunk 1]
```diff
@@ -12,7 +12,18 @@ pub fn sys_mprotect(addr: Vaddr, len: usize, perms: u64, ctx: &Context) -> Resul
...
     let root_vmar = ctx.process.root_vmar();
-    debug_assert!(addr % PAGE_SIZE == 0);
+
+    // According to linux behavior,
+    // <https://elixir.bootlin.com/linux/v6.0.9/source/mm/mprotect.c#L681>,
+    // the addr is checked even if len is 0.
+    if addr % PAGE_SIZE != 0 {
+        return_errno_with_message!(Errno::EINVAL, "the start address should be page aligned");
+    }
+
```
Comment (tatetian) [inline kernel/aster-nix/src/syscall/mprotect.rs:22]: Remove the blank line to make the code more compact

File: kernel/aster-nix/src/syscall/munmap.rs [hunk 2]
```diff
@@ -7,6 +7,15 @@ use crate::prelude::*;
 
 pub fn sys_munmap(addr: Vaddr, len: usize, ctx: &Context) -> Result<SyscallReturn> {
     debug!("addr = 0x{:x}, len = {}", addr, len);
+
+    if addr % PAGE_SIZE != 0 {
+        return_errno_with_message!(Errno::EINVAL, "munmap addr must be page-aligned");
+    }
+
```
Comment (tatetian) [inline kernel/aster-nix/src/syscall/munmap.rs:14]: Remove the blank line to make the code more compact

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**